lib/mysqlDaoQueryHelper.js

```diff
@@ -31,32 +31,35 @@ class MysqlDaoQueryHelper {
 
     /**
      * @param {object} createArgs
+     * @param {boolean} [ignoreOnDuplicateKey] if true, insert will ignore duplicate keys
      * @returns {string}
      */
-    create(createArgs) {
-        const cleanValues = this._cleanAndMapValues(createArgs);
-
-        const sql =
-            Squel.insert().into(this._tableName)
-                .setFields(cleanValues, {dontQuote: true})
-                .toString();
-        return sql;
+    create(createArgs, ignoreOnDuplicateKey) {
+        return this.createBulk([createArgs], ignoreOnDuplicateKey);
     }
 
     /**
      * @param {Array.<object>} createArgsArray
+     * @param {boolean} [ignoreOnDuplicateKey] if true, insert will ignore duplicate keys
      */
-    createBulk(createArgsArray) {
+    createBulk(createArgsArray, ignoreOnDuplicateKey) {
         const cleanValues = [];
         createArgsArray.forEach(createArgs => {
             cleanValues.push(this._cleanAndMapValues(createArgs));
         });
 
-        const sql =
-            Squel.insert().into(this._tableName)
-                .setFieldsRows(cleanValues, {dontQuote: true})
-                .toString();
-        return sql;
+        const sqlBuilder =
+            Squel.useFlavour('mysql').insert().into(this._tableName)
+                .setFieldsRows(cleanValues, {dontQuote: true});
+
+        if (ignoreOnDuplicateKey) {
+            //INSERT ... IGNORE actually ignores ALL errors, not just duplicate key, so best practice is to "update"
+            // a duplicate setting a column to itself
+            const firstField = Object.keys(cleanValues[0])[0];
+            sqlBuilder.onDupUpdate(firstField, firstField, {dontQuote: true})
```

Comment:
Not 100% sure, but we may need to do something similar to what we did for upsertBulk.  We had to do 

``` 
INSERT INTO (col1, col2,...)
VALUES 
  ('val1_1', 'val2_1',...),
  ('val1_2, 'val2_2',...)
ON DUPLICATE KEY UPDATE
  col1 = VALUES(col1),
  col2 = VALUES(col2)
```

The natural col1 = col1 ends up using the value from the first set of insert values or something like that.  Might be worth pinging Jason about it b/c he did upsertBulk.  It looks like this might not be an issue based on the unit test but worth confirming.